addons/mail/static/src/models/thread.js

```diff
@@ -45,6 +45,10 @@ registerModel({
             if ('avatarCacheKey' in data) {
                 data2.avatarCacheKey = data.avatarCacheKey;
             }
+            if ('channel' in data) {
+                data2.channel = data.channel;
+                data2.model = 'mail.channel';
+            }
```

Comment:
note: the channel was specified after model to make sure it was overriding model if somehow necessary